06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js

```diff
@@ -0,0 +1,200 @@
+   * @returns {Object} - returns the file that has been uploaded
+   * @memberOf InvertedIndex
+   */
+  logFile() {
+    return this.uploadedFile;
+  }
+  /**
+   * checkFileExtension (it returns the file extension)
+   * @param {string} uploadedFileName
+   * @return {string} - returns a string of the
+   * file extension of the uploaded file
+   * @memberOf InvertedIndex
+   */
+  static checkFileExtension(uploadedFileName) {
+    const array = uploadedFileName.split('.');
+    return array[array.length - 1];
+  }
+  /**
+   * checkIfJson (it returns true if the file extension is .json)
+   * @param {string} uploadedFileName
+   * @returns {boolean} - returns true if the file extension is json else false
+   * @memberOf InvertedIndex
+   */
+  checkIfJson(uploadedFileName) {
+    const fileExtension = this.checkFileExtension(uploadedFileName);
+    return fileExtension.toLowerCase() === 'json';
+  }
+  /**
+   * containsTitleText (it checks if the content of
+   * the file contains title and text)
+   * @returns {Boolean} - returns true if the file
+   * contains title and text or false if it does not
+   * @memberOf InvertedIndex
+   */
+  containsTitleText() {
+    for (let arrayIndex = 0; arrayIndex < this.uploadedFile.length;
+    arrayIndex += 1) {
+      if (('title' in this.uploadedFile[arrayIndex])
+      && ('text' in this.uploadedFile[arrayIndex])) {
+        return true;
+      }
+    }
+    return false;
+  }
+  /**
+   * getTitlesAndTexts (it gets all the titles and texts in the uploaded file)
+   * @returns {Object} - returns an object with keys titles and texts
+   * @memberOf InvertedIndex
+   */
+  getTitlesAndTexts() {
+    const objectToReturn = [{ titles: [] }, { texts: [] }];
+
+    this.uploadedFile.forEach((e) => {
+      objectToReturn[0].titles.push(e.title.split(':')[0]);
+      objectToReturn[1].texts.push(e.text);
+    });
+
+    return objectToReturn;
+  }
+  /**
+   * getIndexedWords (it returns an array of indexed words)
+   * @returns {Array} - returns an array of indexed words
+   * @memberOf InvertedIndex
+   */
+  getIndexedWords() {
+    let uncleanIndexedWords = '';
+    const fileToIndex = this.getTitlesAndTexts();
+    const arrayIndex = 1;
+
+    fileToIndex[arrayIndex].texts.forEach((e) => {
+      uncleanIndexedWords += `${e} `;
+    });
+
+    const cleanIndexedWords = this.cleanIndexedWords(uncleanIndexedWords);
+    // To remove the empty space at the end of the cleanIndexedWords array
+    cleanIndexedWords.pop();
+    return cleanIndexedWords;
+  }
+  /**
+   * cleanIndexedWords (it returns an array of words without commas,
+   * fullstop, spaces and also unique words)
+   * @param {string} uncleanWords
+   * @returns {Array} - an array of unique words
+   * @memberOf InvertedIndex
+   */
+  static cleanIndexedWords(uncleanWords) {
+    const wordsToLowercase = uncleanWords.toLowerCase();
+    const cleanWords = wordsToLowercase.replace(/\.|,/g, '').split(' ');
```

Comment:
Good work. One Last thing you need to strip special characters and remove trim whitespace.